tests/aws/services/lambda_/event_source_mapping/test_lambda_integration_kinesis.py

```diff
@@ -537,12 +544,222 @@ def test_kinesis_event_source_mapping_with_on_failure_destination_config(
 
         def verify_failure_received():
             result = aws_client.sqs.receive_message(QueueUrl=queue_event_source_mapping)
-            assert result["Messages"]
+            assert result.get("Messages")
             return result
 
-        sqs_payload = retry(verify_failure_received, retries=50, sleep=5, sleep_before=5)
+        sleep = 15 if is_aws_cloud() else 5
+        sqs_payload = retry(verify_failure_received, retries=15, sleep=sleep, sleep_before=5)
         snapshot.match("sqs_payload", sqs_payload)
 
+    @pytest.mark.skipif(
+        is_old_esm(),
+        reason="ReportBatchItemFailures: Partial batch failure handling not implemented in ESM v1",
+    )
+    @markers.snapshot.skip_snapshot_verify(
+        paths=[
+            "$..Messages..Body.KinesisBatchInfo.shardId",
```

Comment:
nit: might be worth commenting why we have a parity mismatch here

Reply:
We do this since the shardId and AWS account ID have a mismatch when transforming due to the same 12 digit string of numbers

https://github.com/localstack/localstack/blob/08a3efd4a2c3c066777273c3e14e73ff0b0e7920/tests/aws/services/lambda_/event_source_mapping/test_lambda_integration_kinesis.py#L560-L561